python/pylint/pylint_main.py

```diff
@@ -0,0 +1,26 @@
+# pylint: skip_file
+
+"""Entrypoint to running pylint_test."""
+
+import os
+import sys
+
+import pylint
+
+from rules_128tech import sharder
+
+
+def main():
+    sep = sys.argv.index("--")
+    if sep == -1:
+        os.exit("expected to find '--' on the command line")
```

Comment:
This might be a little awkward if someone wanted to pass args to pylint via `--test_arg`, but I think that is pretty rare compared to the way we do for pytest, so this is probably fine.

As an alternative could you just exclude all args starting with `-` from the filtering? It's still a little hacky but would allow passthrough args from bazel (except stuff like `--foo bar` but that already gets messy with bazel `--test_arg="--foo bar"` so I usually use something like `--test_arg=--foo=bar` anyway)

Reply:
crap I totally didn't think about `--test_arg`.

yeah, I didn't want to deal with trying to understand flags like `--foo bar`. in bazel 4.0 there is an [`env`](https://docs.bazel.build/versions/master/be/common-definitions.html#test.env) in tests so we could pass in the source files as an environment variable and append them to the command line, but in the meantime I see two possible solutions,
- Convert this file to be a template and substitute in the source files as a python variable,
- add some sentinels maybe `__PYLINT_START_FILES__` and `__PYLINT_END_FILES__` and look for those instead of `"--"`

I'm thinking that the first option would be cleaner